$NetBSD$

Fix RPATH.
Use MACHINE_GNU_PLATFORM for the platform triple.

--- CMakeLists.txt.orig	2024-09-09 19:38:20.000000000 +0000
+++ CMakeLists.txt
@@ -169,6 +169,7 @@ MESSAGE(STATUS "Is big endian ... ${PLAT
 
 TRIPLE(SAGITTARIUS_PROCESSOR SAGITTARIUS_PLATFORM
   SAGITTARIUS_OS SAGITTARIUS_VENDOR SAGITTARIUS_TRIPLE)
+SET(SAGITTARIUS_TRIPLE ${MACHINE_GNU_PLATFORM})
 
 ### System 
 
@@ -224,6 +225,8 @@ IF(WIN32 AND NOT CYGWIN)
   ENDIF()
   SET(WINDOWS_SUPPORT TRUE)
 ELSE()
+  SET(SAGITTARIUS_LIB_PATH
+    "${CMAKE_INSTALL_PREFIX}/${LIB_DIR}")
   SET(SAGITTARIUS_SHARE_LIB_PATH
     "${CMAKE_INSTALL_PREFIX}/${SAGITTARIUS_CURRENT_SHARE}/lib")
   SET(SAGITTARIUS_SHARE_SITE_LIB_PATH 
@@ -237,9 +240,10 @@ ELSE()
 
   # RPATH
   SET(CMAKE_SKIP_BUILD_RPATH FALSE)
-  SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+  SET(CMAKE_SKIP_RPATH FALSE)
+  SET(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
   SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
-  SET(CMAKE_INSTALL_RPATH "${SAGITTARIUS_DYNLIB_PATH}")
+  SET(CMAKE_INSTALL_RPATH "${SAGITTARIUS_LIB_PATH}:${SAGITTARIUS_DYNLIB_PATH}")
 ENDIF()
 
 MARK_AS_ADVANCED(SAGITTARIUS_VERSION SAGITTARIUS_BUGREPORT
